app/hooks/throttled.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {throttle} from 'lodash';
+import {useMemo, useRef} from 'react';
+
+const useThrottled = <T extends any[]>(callback: (...args: T) => void, time: number) => {
+    const callbackRef = useRef(callback);
+
+    callbackRef.current = callback;
+
+    return useMemo(
+        () => throttle((...args: T) => callbackRef.current(...args), time),
+        [time],
+    );
+};
```

Comment:
I like it. It still has a potential issue where changing the time would recreate, and therefore, remove the throttle, but since usually the time is constant, that won't be an issue in most scenarios.

I also like the trick of putting the callback in a ref, so changes in the callback don't change the behavior of this function. Great work!